### PR TITLE
[API] Enable Absolute Local Imports

### LINKED ISSUE #184: Refactor local `imports` to use absolute paths.
## User Story

As a developer I want to use absolute imports so that I don't have to keep track of complex path traversal. 

#### Problem Details _(Optional)_

https://github.com/ApollosProject/apollos-prototype/blob/master/api/src/connectors/rock/Model.js#L1
```
import { createCursor, parseCursor } from '../../utils/cursor';
// should become
import { createCursor, parseCursor } from 'app/utils/cursor';
```

https://spin.atomicobject.com/2017/10/07/absolute-paths-javascript/
https://github.com/facebook/react-native/issues/3099#issuecomment-221815006
https://github.com/tleunen/babel-plugin-module-resolver

#### Acceptance Criteria

- [ ] Eliminate use of `../..` in the API

### PR DESCRIPTION
**Pull request by vinnyjth**
Absolute paths are easier to use than relative paths, no need to try and
guess how many levels up or down you need to traverse to import your
file.

To help distinguish absolute "local" paths from "packages", I have opted for
the convention of prefixing absolute paths with "/app". You can also
choose to omit the "/app" and import directly.

Fixes #184 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings (in app AND in storybook)
- [x] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer
- [x] Add a line to the changelog documenting the high-level change this PR introduces

## Reviewer

- [x] Review CI build results
- [x] Review function and form on iOS
- [x] Review function and form on Android
- [x] Review new stories if any apply
- [x] Review test coverage: Are all new or changed states tested?
- [x] Review new test logic
- [ ] Review the changelog entry

### CODE REVIEW

**Comment by conrad-vanl** [comment]
@burkeshartsis what are your thoughts on this vs what we are doing on `mobile`?

**Comment by vinnyjth** [comment]
@conrad-vanl and @burkeshartsis Would love to hear your thoughts on the `/app/${filename}` (api) vs straight up `${filename}` (mobile) :) 

That said, please keep in mind that there's a technical limitation that will make it difficult to use the `${filename}` strategy on API, the Jest bundler gets really confused at this line: https://github.com/ApollosProject/apollos-prototype/pull/194/files#diff-f766a54d28e5832913c96b4c69625a3aL7

We could figure it out, but it would be easier to force `/app` as a paradigm as long as we do cross project imports :)

**Comment by conrad-vanl** [comment]
I like the idea. I'd want to do the same on `/mobile` as well though so we don't have two competing paradigms

**Comment by vinnyjth** [comment]
@conrad-vanl Sounds great to me! I can do that in this PR :) I'll also rename `/app` to `/api`

**Comment by burkeshartsis** [comment]
I'm in support of these changes given the limitations.

File: api/.eslintrc.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = {
     'react/jsx-filename-extension': 0,
     'react/prefer-stateless-function': [2, { ignorePureComponents: true }],
     'import/no-extraneous-dependencies': 0,
+    'import/no-absolute-path': 0,
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply api/.eslintrc.js]: Absolutely! I say we can leave it ignored until this import linter rule starts working again :)

**Pull request merged**